### PR TITLE
Get Legal Moves Service

### PR DESCRIPTION
**Pull request by lancemcmuffin**
Adds service for getting legal moves

Linked to: https://dev.azure.com/TeamWeThemBoys/xGame/_workitems/edit/99

### CODE REVIEW

File: xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java [hunk 1]
```diff
@@ -104,4 +107,28 @@ private String stringifyBoard() throws JsonProcessingException {
...
+				return legalMoves;
+			}
+
+			System.out.println(piece);
+			legalMoves = piece.legalMoves();
+
+		} catch (IllegalPositionException e1) {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+					"Cannot get legal moves for " + piecePosition + ". Position is illegal.", e1);
+		}
```
Comment (tatunka) [inline xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java:114]: you're already catching this in the controller. Pick one or the other.

File: xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java [hunk 2]
```diff
@@ -104,4 +107,28 @@ private String stringifyBoard() throws JsonProcessingException {
 		mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
 		return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this.board.getBoard());
 	}
+	
+	public List<String> getLegalMoves(int matchId, String piecePosition)
```
Comment (tatunka) [inline xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java:94]: move method to before the private methods.

**Review by tatunka — CHANGES_REQUESTED** [review]
Good work. Just some tiny nitpicks.

File: xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java (hunk 1 above)
Comment (lancemcmuffin) [reply xGame/Java/xgame/src/main/java/com/xgame/service/GameService.java:114]: Removed from controller

**Review by tatunka — APPROVED** [review]

**Review by aguentchev — APPROVED** [review]

**Pull request merged**